File: Java/src/main/java/com/backend/java/controller/AuthController.java
package com.backend.java.controller;

import com.backend.java.model.User;
import com.backend.java.iservice.IAuthService;
import com.backend.java.jwt_token.JwtUtil;
import com.backend.java.iservice.IUserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Optional;

@RestController
@RequestMapping("/api/auth")
public class AuthController {
    @Autowired
    private IAuthService authService;

    @Autowired
    private IUserService userService;

    @Autowired
    private JwtUtil jwtUtil;

    @PostMapping("/register")
    public ResponseEntity<?> registerUser(@RequestBody User user) {
        if (userService.findByUsername(user.getUsername()).isPresent()) {
            return ResponseEntity.badRequest().body("Username is already taken");
        }
        if (userService.findByEmail(user.getEmail()).isPresent()) {
            return ResponseEntity.badRequest().body("Email is already taken");
        }
        User registeredUser = authService.registerUser(user);
        String token = jwtUtil.generateToken(registeredUser.getId());

        return ResponseEntity.ok(token);
    }

    @PostMapping("/login")
    public ResponseEntity<?> loginUser(@RequestBody User user) {
        Optional<User> existingUser = userService.findByUsername(user.getUsername());
        if (existingUser.isPresent() && authService.validatePassword(user.getPasswordHash(), existingUser.get().getPasswordHash())) {
            String token = jwtUtil.generateToken(existingUser.get().getId());
            return ResponseEntity.ok(token);
        } else {
            return ResponseEntity.status(401).body("Invalid username or password");
        }
    }

}

File: Csharp/WebApiSN/Controllers/AuthController.cs
namespace WebApiSN.Controllers;
using IServices;
using Models.UserManagement;
using Models;
using Microsoft.AspNetCore.Mvc;
using System.Threading.Tasks;


    [ApiController]
    [Route("api/[controller]")]
    public class AuthController : ControllerBase
    {
        private readonly IAuthService _authService;

        public AuthController(IAuthService authService)
        {
            _authService = authService;
        }

        [HttpPost("register")]
        public async Task<IActionResult> RegisterUser([FromBody] RegisterUser user)
        {
            var token = await _authService.RegisterUserAsync(user);
            return Ok(token);
        }

        [HttpPost("login")]
        public async Task<IActionResult> LoginUser([FromBody] LoginUser user)
        {
            var token = await _authService.LoginUserAsync(user);
            if (token != null)
            {
                return Ok(token);
            }
            return Unauthorized("Invalid username or password");
        }
    }

